reframe/core/pipeline.py

```diff
@@ -957,11 +957,16 @@ def supports_system(self, partition_name):
             return True
 
         # Check if this is a relative name
-        if partition_name.find(':') == -1:
-            partition_name = '%s:%s' % (self.current_system.name,
-                                        partition_name)
+        if partition_name.find(':') != -1:
+            system_partition, partition_name = partition_name.split(':')
+        else:
+            system_name = self.current_system.name
+            partition_name = partition_name
+
+        specific = '%s:%s' % (system_name, partition_name)
+        generic = '*:%s' % partition_name
 
-        return partition_name in self.valid_systems
+        return specific in self.valid_systems or generic in self.valid_systems
```

Comment:
First, I would change the formal argument name to something more generic: `partition_name` is a bit misleading. I would suggest to use just `name`. Second, you could avoid the hassle of checking the different formats if we had a `normalize_partname()` function (even nested inside `supports_system()`) that would normalize the the partition names and then try to match them. Something along the following lines:

```python
    def supports_system(self, name):
        def normalize_partname(name):
            pass

        def match_system(patt, normalized_name):
            pass

        return any(match_system(normalize_partname(n), normalize_partname(name)) for n in self.valid_systems)
```

The `match_system()` would do the simple matching we are looking for. We don't want neither regex matches nor fnmatches.

Reply:
implemented differently, but should achieve the same idea